src/components/domain/createLuckyDay/selectActivity/container/activityToggle/hooks/useCustomInput.ts

```diff
@@ -0,0 +1,72 @@
+import React, { useRef, useState } from "react";
+import type { UseFormSetValue, UseFormWatch } from "react-hook-form";
+
+import { useToast } from "hooks";
+import type { CreateLuckyDayForm } from "types";
+
+interface useCustomInputProps {
+  setValue: UseFormSetValue<CreateLuckyDayForm>;
+  watch: UseFormWatch<CreateLuckyDayForm>;
+}
+
+const useCustomInput = ({ setValue, watch }: useCustomInputProps) => {
+  const [text, setText] = useState("");
+
+  const spanRef = useRef<HTMLSpanElement>(null);
+
+  const inputWidth = text.length
+    ? spanRef.current?.getBoundingClientRect().width
+    : 0;
+  const { addToast } = useToast();
+
+  const handleCustomItemChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target.value.length > 14) return;
+
+    setText(e.target.value);
+  };
+
+  const handleAddCustomActivity = (e: React.MouseEvent): void => {
+    e.stopPropagation();
+
+    const checkSameActivity = watch("customActList")?.includes(text);
+
+    if (checkSameActivity) {
+      addToast({ content: "이미 추가된 활동입니다." });
+      setText("");
+
+      return;
+    }
+
+    setValue("customActList", [...(watch("customActList") ?? ""), text]);
+    setText("");
+  };
+
+  const handleEnterCustomItemChange = (
+    e: React.KeyboardEvent<HTMLInputElement>
+  ) => {
+    if (e.key === "Enter") {
```

Comment:
여기 영어나 버튼으로 추가할 땐 문제가 없는데 한글을 enter로 추가하면 마지막 글자로 태그가 하나 더 생기는 버그가 있는 것 같습니다.

https://github.com/user-attachments/assets/a6960de4-45fb-4b13-83af-780c8331cc27

한글 IME 입력 방식 때문인 것 같은데 아래처럼 수정하면 해결되는 것 같습니다. 

```suggestion
    if (e.key === "Enter" && !e.nativeEvent.isComposing) {
```

https://github.com/user-attachments/assets/b703c4ec-b35d-4ecc-a1b0-260463dec835

Reply:
수정한줄 알았는데 누락됐던 것 같아요 ㅠㅠ 수정했습니다.